lib/modules/datasource/nuget/v3.ts

```diff
@@ -242,3 +267,65 @@ export async function getReleases(
 
   return dep;
 }
+
+async function getSourceUrlFromNupkgCached(
+  http: Http,
+  packageName: string,
+  packageVersion: string | null,
+  nupkgUrl: string,
+): Promise<string | null> {
+  const cacheKey = `source-url:${nupkgUrl}`;
```

Comment:
Does this URL change every time there's a new release?

I would like it to be cached per _package_ for one week

Reply:
yes, that url always includes the version

Reply:
we should also do the caching for the nuspec file, which currently also occur on every getReleases call.

Reply:
So the cacheKey should be changed to be based off the registryUrl+packageName, not nupkgUrl

Reply:
I am not sure about just taking the package name. It's not too uncommon that a source project changes over time. Depending on which version we look at first we might end up with outdated source URLs being cached for quite some time.

Reply:
My experience is that packages in most ecosystems very rarely change source URLs. Most commonly never (i.e. same github repo forever), sometimes 1-2 changes over years, almost never ever "regularly". Is it different for nuget?

Reply:
I have seen it from time to time, especially with Microsoft/.NET packages. But we can also start with just `registryUrl+packageName` and see how it behaves. Would be easy to adapt if needed.

Reply:
Adapted in ca67215479147747d209ec929afbaa950cb0e6b5